build

```diff
@@ -203,6 +203,29 @@ function do_push() {
     done
 }
 
+function do_tag() {
+    local tag token
+    TAG_PREFIX=
+    echo "Pulling latest images"
+    for tag in base latest "${BASE_VARIANTS[@]}" "${VARIANTS[@]}"; do
+        tag="${tag,,}"
+        tag="$(image_name "${tag}")"
+        docker pull "$tag"
+    done
+    token="$(curl "https://ghcr.io/token?scope=repository:datadog/dd-trace-java-docker-build:pull" | jq -r .token)"
+    latest_version="$(curl -s -k -X GET -H "Authorization: Bearer $token" -H "Accept: application/json" "https://ghcr.io/v2/datadog/dd-trace-java-docker-build/tags/list" | jq -r '.tags[]' | grep ^v | sed -e 's~^v~~g' -e 's~-.*~~g' | sort -rn | head -n 1)"
+    latest_version="${latest_version:-0}"
+    next_version="$((latest_version+1))"
```

Comment:
Wooh, seems pretty complicated. 
Are you confident it worth the trouble rather than a date-based versioning? (ie 23.09 for example)

Reply:
Fair. Changed to `v23.09`.